addons/account/models/account_payment.py

```diff
@@ -186,585 +289,431 @@ def _get_method_codes_needing_bank_account(self):
+                if len(liquidity_lines) != 1 or len(counterpart_lines) != 1:
+                    raise UserError(_(
+                        "The journal entry %s reached an invalid state regarding its related payment.\n"
+                        "To be consistent, the journal entry must always contains:\n"
+                        "- one journal item involving the outstanding payment/receipts account.\n"
+                        "- one journal item involving a receivable/payable account.\n"
+                        "- optional journal items, all sharing the same account.\n\n"
+                    ) % move.display_name)
+
+                if writeoff_lines and len(writeoff_lines.account_id) != 1:
+                    raise UserError(_(
+                        "The journal entry %s reached an invalid state regarding its related payment.\n"
+                        "To be consistent, all the write-off journal items must share the same account."
+                    ) % move.display_name)
+
+                if any(line.currency_id != all_lines[0].currency_id for line in all_lines):
+                    raise UserError(_(
+                        "The journal entry %s reached an invalid state regarding its related payment.\n"
+                        "To be consistent, the journal items must share the same currency."
+                    ) % move.display_name)
+
+                if any(line.partner_id != all_lines[0].partner_id for line in all_lines):
+                    raise UserError(_(
+                        "The journal entry %s reached an invalid state regarding its related payment.\n"
+                        "To be consistent, the journal items must share the same partner."
+                    ) % move.display_name)
+
+                if counterpart_lines.account_id.user_type_id.type == 'receivable':
+                    partner_type = 'customer'
+                else:
+                    partner_type = 'supplier'
+
+                liquidity_amount = liquidity_lines.amount_currency if liquidity_lines.currency_id else liquidity_lines.balance
+
+                move_vals_to_write.update({
+                    'currency_id': liquidity_lines.currency_id.id or liquidity_lines.company_currency_id.id,
+                    'partner_id': liquidity_lines.partner_id.id,
+                })
+                payment_vals_to_write.update({
+                    'amount': abs(liquidity_amount),
+                    'payment_type': 'inbound' if liquidity_amount > 0.0 else 'outbound',
+                    'payment_reference': liquidity_lines.name,
+                    'partner_type': partner_type,
+                    'currency_id': liquidity_lines.currency_id.id or liquidity_lines.company_currency_id.id,
+                    'destination_account_id': counterpart_lines.account_id.id,
+                    'partner_id': liquidity_lines.partner_id.id,
+                })
+
+            move.write(move._cleanup_write_orm_values(move, move_vals_to_write))
```

Comment:
this is because `partner_id` & `currency_id` are duplicated on both models and then, we need to synchronize them. This should be the case in the future but this is really hard (I tried) with the current design of account.move...